File: UserControlForm/UserControlForm/UserControlForm.Web/Modules/Integration/Movie/MovieColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace UserControlForm.Integration.Columns;

[ColumnsScript("Integration.Movie")]
[BasedOnRow(typeof(MovieRow), CheckNames = true)]
public class MovieColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public int Type { get; set; }
    public string Description { get; set; }
}

File: UserControlForm/UserControlForm/UserControlForm.Web/Modules/ServerTypes/Integration/MovieColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { MovieRow } from "./MovieRow";

export interface MovieColumns {
    Id: Column<MovieRow>;
    Name: Column<MovieRow>;
    Type: Column<MovieRow>;
    Description: Column<MovieRow>;
}

export class MovieColumns extends ColumnsBase<MovieRow> {
    static readonly columnsKey = 'Integration.Movie';
    static readonly Fields = fieldsProxy<MovieColumns>();
}